src/browser/fetch/header.rs

```diff
@@ -0,0 +1,54 @@
+//! HTTP headers
+
+use std::borrow::Cow;
+
+/// Request headers.
+#[derive(Clone, Debug, Default)]
+pub struct Headers<'a>(Vec<Header<'a>>);
+
+impl<'a> Headers<'a> {
+    /// Sets a new value for an existing header or adds the header if
+    /// it does not already exist.
+    pub fn set(&mut self, header: Header<'a>) {
+        self.0.retain(|Header { name, .. }| &header.name != name);
+        self.0.push(header);
+    }
+}
+
+impl<'a> IntoIterator for Headers<'a> {
+    type Item = Header<'a>;
+    type IntoIter = std::vec::IntoIter<Self::Item>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.0.into_iter()
+    }
+}
```

Comment:
Why we need `Headers` struct? And why it contains `Vec`?
Can we replace it with `BTreeMap<Header>`?
  - It already implements iterators.
  - We don't need to use weird combination of `self.0.retain` + `push`.
  - It would be ordered in logs.

Reply:
Mostly for performance reasons. Headers most of the time are very small, thus using hashmaps/trees for tiny number of entries has a worse performance, despite o(1) vs o(n).

Eventually we will also probably add [append method](https://developer.mozilla.org/en-US/docs/Web/API/Headers/append), so separate struct is desirable.

Reply:
> Eventually we will also probably add append method, so separate struct is desirable.

Then. I would rather modify `Header` - change its field `value` to `values: Vec<..>`.

> Mostly for performance reasons.

We lose some performance in `Header::set` with `Vec` so I don't think there will be a difference. And I think it's again premature optimization.